modules/core/src/test/java/org/apache/ignite/internal/processors/cache/GridTransactionsSystemUserTimeMetricsTest.java

```diff
@@ -354,10 +367,38 @@ private ClientTxTestResult measureClientTransaction(boolean systemDelay, boolean
      *
      * @param histogramJson Histogram json.
      */
-    private void assertNotEmpty(String histogramJson) {
+    private void checkHistogram(String histogramJson, int txCnt) {
         assertNotNull(histogramJson);
 
-        assertTrue(histogramJson.length() > 0);
+        assertTrue(histogramJson, histogramJson.matches(HISTOGRAM_REGEX));
+        assertTrue(histogramJson, histogramJson.contains("\"values\":[{\"fromInclusive\":0,\"toInclusive\":1,\""));
+
+        Matcher m = HISTOGRAM_VALUE_COUNT_PATTERN.matcher(histogramJson);
+
+        int cnt = 0;
+
+        while (m.find()) {
+            String c = m.group("count");
+
+            cnt += Integer.valueOf(c);
+        }
+
+        assertEquals("Must be " + txCnt + " transaction(s), actually were: " + cnt + ". Histogram: " + histogramJson, txCnt, cnt);
+    }
+
+    /**
+     * Creates histogram regex to check transaction histograms.
+     * @return Regex.
+     */
+    private static String histogramRegex() {
+        String regex = toJson(new HistogramMetric(METRIC_TIME_BUCKETS))
```

Comment:
Looks like we can just return the result of `toJson`. I don't a reason for creating temp variable.